app/controllers/green_lanes/results_controller.rb

```diff
@@ -2,66 +2,67 @@ module GreenLanes
-        results_params[:commodity_code],
-        { filter: {
-          geographical_area_id: results_params[:country_of_origin],
-          moving_date: results_params[:moving_date],
-        } },
-        { authorization: TradeTariffFrontend.green_lanes_api_token },
-      )
+      @next_page = determine_next_page
+    end
+
+    private
+
+    def set_up
+      check_green_lanes_enabled
+      disable_switch_service_banner
+      disable_search_form
+      goods_nomenclature
+      set_instance_variables
+      determine_assessments
+    end
+
+    def results_params
+      params.permit(:commodity_code, :country_of_origin, :moving_date, :c1ex, :c2ex, :category, :ans)
+    end
+
+    def goods_nomenclature
+      @goods_nomenclature ||= GreenLanes::FetchGoodsNomenclature.new(results_params).call
+    end
 
-      @commodity_code = goods_nomenclature.goods_nomenclature_item_id
+    def set_instance_variables
+      @commodity_code = @goods_nomenclature.goods_nomenclature_item_id
       @country_of_origin = results_params[:country_of_origin] || GeographicalArea::ERGA_OMNES
       @country_description = GeographicalArea.find(@country_of_origin).description
       @moving_date = results_params[:moving_date]
-
-      next_page = DetermineNextPage
-        .new(goods_nomenclature)
-        .next(
-          cat_1_exemptions_apply:,
-          cat_2_exemptions_apply:,
-        )
-
-      case next_page
-      when :result_cat_1
-        render 'result_cat_1'
-      when :result_cat_2
-        render 'result_cat_2'
-      when :result_cat_3
-        render 'result_cat_3'
-      end
+      @category = results_params[:category]
+      @ans = JSON.parse(results_params[:ans])
     end
 
-    private
+    def determine_assessments
+      assessments = GreenLanes::DetermineAssessments.new(@goods_nomenclature, @ans)
+      @category_1_assessments = assessments.category_1_assessments
+      @category_2_assessments = assessments.category_2_assessments
+      @no_cat1_exemptions = assessments.no_cat1_exemptions
+      @no_cat2_exemptions = assessments.no_cat2_exemptions
+      @no_exemptions = assessments.no_exemptions
+      @cat_1_exemptions = assessments.cat_1_exemptions
+      @cat_2_exemptions = assessments.cat_2_exemptions
+      @cat_1_exemptions_met = assessments.cat_1_exemptions_met
+      @cat_2_exemptions_met = assessments.cat_2_exemptions_met
+      @no_exemptions_met = assessments.no_exemptions_met
+    end
 
-    def results_params
-      params.permit(
-        :commodity_code,
-        :country_of_origin,
-        :moving_date,
-        :c1ex,
-        :c2ex,
+    def determine_next_page
+      DetermineNextPage.new(@goods_nomenclature).next(
+        cat_1_exemptions_apply:,
+        cat_2_exemptions_apply:,
       )
     end
 
     def cat_1_exemptions_apply
-      case results_params[:c1ex]
-      when 'true' then true
-      when 'false' then false
-      end
+      results_params[:c1ex] == 'true'
```

Comment:
I'm wondering what happens, here, if there are no cat_1_exemptions questions and we pass the answer as false?

```ruby
{
  c1ex: nil,
  c2ex: nil,
}

cat_1_exemptions_apply # false
cat_2_exemptions_apply # false
```

Reply:
Yeah i did see this but then its caught further down the line as they wont have the cat 2 answers at all in the params?

the {1:{...},2:{...}}

Reply:
Maybe we can just pass the raw value to the service, then